clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheEntityResponse.java

```diff
@@ -82,10 +113,14 @@ public static Failure failure(Exception cause) {
 
     private final Exception cause;
 
-    private Failure(Exception cause) {
+    Failure(Exception cause) {
       this.cause = cause;
     }
 
+    Failure(ByteBuffer failure) {
+      this.cause = (Exception)Util.unmarshall(failure.array());
```

Comment:
AFAICT this constructor is package local and is only being called by providing `ByteBuffer.wrap()` which uses `HeapByteBuffer`. So it should be fine.